sw/device/silicon_creator/lib/sigverify/sphincsplus/params.h

```diff
@@ -0,0 +1,149 @@
+   */
+  kSpxForsTrees = 14,
+  /**
+   * Winternitz parameter.
+   */
+  kSpxWotsW = 16,
+  /**
+   * Number of bytes in a hypertree address (for clarity).
+   */
+  kSpxAddrBytes = 32,
+  /**
+   * Bit-length of the Winternitz parameter.
+   */
+  kSpxWotsLogW = 4,
+  /**
+   * Parameter `len1` for WOTS signatures.
+   *
+   * See section 3.1 of the SPHINCS+ NIST submission:
+   *   https://sphincs.org/data/sphincs+-r3.1-specification.pdf
+   */
+  kSpxWotsLen1 = (8 * kSpxN) / kSpxWotsLogW,
+  /**
+   * Parameter `len2` for WOTS signatures.
+   *
+   * This value is precomputed and equal to:
+   *   floor(log(len1 * (w - 1)) / log(w)) + 1
+   *
+   * During the WOTS computation, the maximum checksum value is `len1 * (w -
+   * 1)`. The log() functions here essentially apply the change-of-base rule;
+   * what we are actually computing is floor(log_w(len1 * (w - 1))) + 1, which
+   * expresses the number of base-w integers required to encode the checksum.
+   *
+   * Some precomputed values based on w and n:
+   *   +------+----------------+----------+
+   *   |  w   |        n       |   len2   |
+   *   +------+----------------+----------+
+   *   | 256  |   0 < n <= 1   |    1     |
+   *   | 256  |   1 < n <= 256 |    2     |
+   *   |  16  |   0 < n <= 8   |    2     |
+   *   |  16  |   8 < n <= 136 |    3     |
+   *   |  16  | 136 < n <= 256 |    4     |
+   *   +------+----------------+----------+
+   *
+   * See section 3.1 of the SPHINCS+ NIST submission:
+   *   https://sphincs.org/data/sphincs+-r3.1-specification.pdf
+   */
+  kSpxWotsLen2 = 4,
+  /**
+   * Number of chains to compute for a WOTS signature.
+   */
+  kSpxWotsLen = kSpxWotsLen1 + kSpxWotsLen2,
+  /**
+   * WOTS signature length in bytes.
+   *
+   * The signature is composed of `kSpxWotsLen` blocks of `kSpxN` bytes each.
+   */
+  kSpxWotsBytes = kSpxWotsLen * kSpxN,
+  /**
+   * WOTS public key length in bytes.
+   */
+  kSpxWotsPkBytes = kSpxWotsBytes,
+  /**
+   * Subtree size.
+   */
+  kSpxTreeHeight = kSpxFullHeight / kSpxD,
+  /**
+   * FORS message length.
+   */
+  kSpxForsMsgBytes = ((kSpxForsHeight * kSpxForsTrees) + 7) / 8,
+  /**
+   * FORS signature length.
+   */
+  kSpxForsBytes = (kSpxForsHeight + 1) * kSpxForsTrees * kSpxN,
+  /**
+   * FORS public key length.
+   */
+  kSpxForsPkBytes = kSpxForsBytes,
+  /**
+   * SPHINCS+ signature.
+   */
+  kSpxBytes =
+      kSpxN + kSpxForsBytes + kSpxD * kSpxWotsBytes + kSpxFullHeight * kSpxN,
+  /**
+   * SPHINCS+ public key length.
+   */
+  kSpxPkBytes = 2 * kSpxN,
+  /**
+   * SPHINCS+ secret key length.
+   */
+  kSpxSkBytes = 2 * kSpxN + kSpxPkBytes,
+  /**
+   * Hash output length (n) in words.
+   */
+  kSpxNWords = kSpxN / sizeof(uint32_t),
+  /**
+   * WOTS signature size in words.
+   */
+  kSpxWotsWords = kSpxWotsBytes / sizeof(uint32_t),
+  /**
+   * WOTS public key size in words.
+   */
+  kSpxWotsPkWords = kSpxWotsPkBytes / sizeof(uint32_t),
+};
+
+#include "shake_offsets.h"
```

Comment:
nit: we generally include files at the top except for templates/x-macros. Can we move the constants in `shake_offsets.h` to this file? We can keep them in a separate `enum` block if you want to keep them separate for clarity (they seem to be originating from different files in [sphincs/sphincsplus](https://github.com/sphincs/sphincsplus)).